### ISSUE TITLE
syncbn cannot support hybridization 

### ISSUE THREAD

**Issue by sunke123** [dmlc/gluon-cv#1106]
Hi, I found that the training for segmentation model does not adopt the hybrid mode (e.g. net.hybridize()).

I tried to use hybrid mode to reduce the training time and found that SyncBN may not support the hybrid mode as described in the official document. When changing the SyncBN to normal BatchNorm, the problem is solved.

Is there something I have missed?

**Comment by bryanyzhu** [comment]
Yes, there is no hybrid mode in training segmentation model. I didn't try it before actually. Can I know what is the error when you run both syncbn and hybrid?

**Comment by bryanyzhu** [comment]
I recently dig into the segmentation code, and you are right. Training for segmentation model does not support the hybrid mode. The reason is because we need to use syncbn and [dataparallel](https://github.com/dmlc/gluon-cv/blob/3bbdc0ada3022f29959ad54b37aa424229906a6f/gluoncv/utils/parallel.py#L138) to make sure segmentation training is stable under small batch size (e.g., bs=2). 

Our implementation of dataparallel depends on mutlitread, however, MXNet is not thread-safe. Hence, you will encounter error when you use hybridize and multithread at the same time.

Right now, I suggest to train the model in non-hybridize way.

**Comment by sunke123** [comment]
@bryanyzhu Thanks for your help.
